src/main/perl/RuleBasedEditor.pm

```diff
@@ -0,0 +1,1175 @@
+# ${developer-info}
+# ${author-info}
+# ${build-info}
+
+=pod
+
+=head1 DESCRIPTION
+
+This module implements a rule-based editor that is used to modify the content
+of an existing file. Each rule driving the editing process is applied to all 
+lines wose "keyword" is matching the one specified in the rule. The input for
+updating the file is a hash typically built from the Quattor configuration when
+the rule-based editor is called from a configuration module. Conditions can be defined
+based on the contents of this configuration. Lines in the configuration file
+that don't match any rule are kept unmodified.
+
+This module is a subclass of the L<CAF::FileEditor>: it extends the base methods of 
+the L<CAF::FileEditor>. It has only one public method (it uses the L<CAF::FileEditor> constructor).
+The methods provided in this module can be combined with L<CAF::FileEditor>
+methods to edit a file.
+
+Rules used to edit the file are defined in a hash: each entry (key/value pair) defines a rule.
+Multiple rules can be applied to the same file: it is important that they are
+orthogonal, else the result is unpredictable. The order used to apply rules is the alphabetical
+order of keywords. Applying the rules to the same configuration always give the same result
+but the changes are not necessarily idempotent (order in which successive edits occured
+may matter, depending on the actual rules).
+
+The hash entry key represents the line keyword in configuration file and
+hash value is the parsing rule for the keyword value. Parsing rule format is :
+
+      [condition->]option_name:option_set[,option_set,...];line_fmt[;value_fmt[:value_fmt_opt]]
+
+If the line keyword (hash key) starts with a '-', the matching
+configuration line will be removed/commented out (instead of added/updated) from the
+configuration file if present. If it starts with a '?', the
+matching line will be removed/commented out if the option is undefined.
+
+=over
+
+=item condition
+
+An option or an option set (see below) that must exist for the rule to be applied.
+Both C<option_set> and C<option_name:option_set> are accepted.
+One option set only is allowed and only its existence (not its value) is tested.
+It is possible to negate the condition (option or option_set must not exist) 
+by prepending it with '!'.
+
+=item option_name 
+
+The name of an option that will be retrieved from the configuration. An option is
+a key in the option set hash.
+
+=item option_set 
+
+The name of an option set where the option is located in (for example 'dpnsHost:dpm'
+means C<dpnsHost> option of C<dpm> option set). An option set is a sub-hash in the configuration
+hash. C<GLOBAL> is a special value for C<option_set> indicating that the option is a global option,
+instead of belonging to a specific option set (global options are at the top level of the configuration
+hash).
+
+=item line_fmt 
+
+Defines the format used to represent the key/value pair. The following formats are
```

Comment:
As there is no duplication of the constant names (see previous comment), I think it helps rather than the opposite to have this list.